### PR TITLE
feat: Integration Testing

### LINKED ISSUE #164: DDK Integration Testing
Implement internal integration tests for DDK Stages.

### PR DESCRIPTION
**Pull request by malachi-constant**
## Details
- Starts #164 
- This PR is meant as a proof-of-concept on building out integration tests. If we move forward with this strategy we'll open a PR to write the remaining necessary tests for #164.
- I'd also recommend we pull our repo internally and run all integration tests against internal accounts so we can avoid configuring credentials with extensive permissions in GIthub.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
Interesting that `yarn.lock` update is no longer complaining. RE: #232 @LeonLuttenberger

**Review by LeonLuttenberger — CHANGES_REQUESTED** [review]
Overrall looks good! I think this is a good approach.

I wonder if we can make this run as part of our GitHub workflow somehow.

File: integ/integ.s3-event.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import * as integration from "@aws-cdk/integ-tests-alpha";
```
[the remark this answers is not in the document]
Comment (malachi-constant) [reply integ/integ.s3-event.ts]: Thanks

File: integ/integ.kinesis-s3-stage.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+class FirehoseToS3StageTestStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props: FirehoseToS3StageTestStackProps) {
+    super(scope, id, props);
+
+    const bucket = new s3.Bucket(this, "Bucket");
+    new FirehoseToS3Stage(this, "Stage", {
+      s3BucketProps: {
+        bucketName: bucket.bucketName,
+        enforceSSL: props.enforceSSL,
+      },
```
[the remark this answers is not in the document]
Comment (malachi-constant) [reply integ/integ.kinesis-s3-stage.test.ts]: Good catch, updated

**Review by LeonLuttenberger — APPROVED** [review]

**Pull request merged**